src/main/java/com/force/i18n/BaseLocalizer.java

```diff
@@ -833,8 +850,14 @@ public DateFormat getInputLongDateFormat() {
      */
     public static DateFormat getLocaleInputDateFormat(Locale locale, TimeZone tz) {
         DateFormat df = getFormatProvider(locale).getDateInstance(DateFormat.SHORT, locale);
+        // ICU-21301 hack: Apply only with an ICU object and when timezone has DST
```

Comment:
according to [ICU-21301](https://unicode-org.atlassian.net/browse/ICU-21301), this only occurs when `setLenient(boolean)` is set to `false`